docs(react-scheduler): add the TypeScript guide (#2278)
DevExpress/devextreme-reactive: packages/dx-react-scheduler/docs/guides/typescript.md

> # React Scheduler - TypeScript

Reviewer: ```suggestion
# React Scheduler - TypeScript Support
```

> [TypeScript](https://www.typescriptlang.org) is a popular JavaScript superset with a strict syntax, developed by Microsoft. It supports static typing and enables catching errors at build time.

Reviewer: ```suggestion

```

> The React Scheduler comes with TypeScript definitions that allows you to use it into TypeScript projects.

Reviewer: ```suggestion
The React Scheduler is supplied with TypeScript definitions. Strict typing allows you to catch bugs at earlier stages and use features like code completion and automated refactoring.
```
